cni/pkg/install-cni/pkg/install/install.go

```diff
@@ -0,0 +1,266 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package install
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	"github.com/coreos/etcd/pkg/fileutil"
+	"github.com/fsnotify/fsnotify"
+	"github.com/pkg/errors"
+
+	"istio.io/istio/cni/pkg/install-cni/pkg/config"
+	"istio.io/istio/cni/pkg/install-cni/pkg/constants"
+	"istio.io/istio/cni/pkg/install-cni/pkg/util"
+	"istio.io/pkg/log"
+)
+
+type configFiles struct {
+	kubeconfigFilepath string
+	cniConfigFilepath  string
+}
+
+// Run starts the installation process with given configuration
+func Run(cfg *config.Config) (err error) {
+	var files configFiles
+
+	ctx, cancel := context.WithCancel(context.Background())
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+	go func(sigChan chan os.Signal, cancel context.CancelFunc) {
+		sig := <-sigChan
+		log.Infof("Exit signal received: %s", sig)
+		cancel()
+	}(sigChan, cancel)
+
+	defer func() {
+		if cleanErr := cleanup(cfg, files); cleanErr != nil {
+			if err != nil {
+				err = errors.Wrap(err, cleanErr.Error())
+			} else {
+				err = cleanErr
+			}
+		}
+	}()
+
+	for {
+		if err = copyBinaries(cfg.UpdateCNIBinaries, cfg.SkipCNIBinaries); err != nil {
+			return
+		}
+
+		var saToken string
+		saToken, err = readServiceAccountToken()
+		if err != nil {
+			return
+		}
+
+		files.kubeconfigFilepath, err = createKubeconfigFile(cfg, saToken)
+		if err != nil {
+			return
+		}
+
+		files.cniConfigFilepath, err = createCNIConfigFile(ctx, cfg, saToken)
+		if err != nil {
+			if err == context.Canceled {
```

Comment:
You could use [the new features from Golang 1.13](https://blog.golang.org/go1.13-errors):
```suggestion
			if errors.Is(err, context.Canceled) {
```